{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "- Neovim version: v0.1.3-355-g6b22a74\n- Operating system: ArchLinux x86_64\n- Terminal emulator: gnome-terminal\n### Actual behaviour\n\nAfter quitting insert mode for first time in the session I get this error:\n\n```\nError detected while processing function provider#python#Call:\nline   18:\nChannel was closed by the client\n```\n\nSubsequent times exiting insert mode throw a different error:\n\n```\nError detected while processing function provider#python#Call:\nline   18:\nInvalid channel \"1\"\n```\n### Expected behaviour\n\nExiting insert mode should work just fine, or at least throw a more helpful error if there is an issue. Line 18 where? Not anywhere in my `.vim` directory.\n### Steps to reproduce\n\nI haven't been able to reproduce this with a minimal configuration yet.\n",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "CONTRIBUTOR",
            "body": "https://github.com/neovim/neovim/issues/4469\n\nSame problem?\n",
            "created_at": "2016-03-19T02:14:23Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-198615734",
            "id": 198615734,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDE5ODYxNTczNA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/198615734/reactions"
            },
            "updated_at": "2016-03-19T02:14:23Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/198615734",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/41495?v=4",
                "events_url": "https://api.github.com/users/Shougo/events{/privacy}",
                "followers_url": "https://api.github.com/users/Shougo/followers",
                "following_url": "https://api.github.com/users/Shougo/following{/other_user}",
                "gists_url": "https://api.github.com/users/Shougo/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/Shougo",
                "id": 41495,
                "login": "Shougo",
                "node_id": "MDQ6VXNlcjQxNDk1",
                "organizations_url": "https://api.github.com/users/Shougo/orgs",
                "received_events_url": "https://api.github.com/users/Shougo/received_events",
                "repos_url": "https://api.github.com/users/Shougo/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/Shougo/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/Shougo/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/Shougo",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I'm not sure. My `.nvimlog` isn't very specific.\n\n```\n2016/03/19 05:02:47 [info @ main:533] 2277 - Starting Neovim main loop.\n2016/03/19 05:02:49 [error @ call_set_error:743] 2277 - msgpack-rpc: Channel was closed by the client\n```\n",
            "created_at": "2016-03-19T04:04:57Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-198630285",
            "id": 198630285,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDE5ODYzMDI4NQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/198630285/reactions"
            },
            "updated_at": "2016-03-19T04:04:57Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/198630285",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1002436?v=4",
                "events_url": "https://api.github.com/users/ntrrgc/events{/privacy}",
                "followers_url": "https://api.github.com/users/ntrrgc/followers",
                "following_url": "https://api.github.com/users/ntrrgc/following{/other_user}",
                "gists_url": "https://api.github.com/users/ntrrgc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ntrrgc",
                "id": 1002436,
                "login": "ntrrgc",
                "node_id": "MDQ6VXNlcjEwMDI0MzY=",
                "organizations_url": "https://api.github.com/users/ntrrgc/orgs",
                "received_events_url": "https://api.github.com/users/ntrrgc/received_events",
                "repos_url": "https://api.github.com/users/ntrrgc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ntrrgc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ntrrgc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ntrrgc",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I believe I'm experiencing the same issue.\n\nIf I set `$NVIM_PYTHON_LOG_FILE`, I get the following message when starting Nvim:\n`2016-05-10 13:13:42,267 [ERROR @ session.py:run:117] 3582 - Setup error: No module named script_host\n`\n",
            "created_at": "2016-05-10T16:06:35Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-218207057",
            "id": 218207057,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxODIwNzA1Nw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218207057/reactions"
            },
            "updated_at": "2016-05-10T16:16:46Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218207057",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/6536964?v=4",
                "events_url": "https://api.github.com/users/vccortez/events{/privacy}",
                "followers_url": "https://api.github.com/users/vccortez/followers",
                "following_url": "https://api.github.com/users/vccortez/following{/other_user}",
                "gists_url": "https://api.github.com/users/vccortez/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/vccortez",
                "id": 6536964,
                "login": "vccortez",
                "node_id": "MDQ6VXNlcjY1MzY5NjQ=",
                "organizations_url": "https://api.github.com/users/vccortez/orgs",
                "received_events_url": "https://api.github.com/users/vccortez/received_events",
                "repos_url": "https://api.github.com/users/vccortez/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/vccortez/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/vccortez/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/vccortez",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I hope to adress the unhelpful message issue with #4723.\n\n@vekat what versions of neovim and the python module (`pip2`/`pip3 show neovim`) do you have?\n",
            "created_at": "2016-05-10T16:21:25Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-218211531",
            "id": 218211531,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxODIxMTUzMQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218211531/reactions"
            },
            "updated_at": "2016-05-10T16:21:25Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218211531",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1363104?v=4",
                "events_url": "https://api.github.com/users/bfredl/events{/privacy}",
                "followers_url": "https://api.github.com/users/bfredl/followers",
                "following_url": "https://api.github.com/users/bfredl/following{/other_user}",
                "gists_url": "https://api.github.com/users/bfredl/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/bfredl",
                "id": 1363104,
                "login": "bfredl",
                "node_id": "MDQ6VXNlcjEzNjMxMDQ=",
                "organizations_url": "https://api.github.com/users/bfredl/orgs",
                "received_events_url": "https://api.github.com/users/bfredl/received_events",
                "repos_url": "https://api.github.com/users/bfredl/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/bfredl/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/bfredl/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/bfredl",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "@bfredl neovim version is 0.1.5-dev, and pip3 shows version 0.1.8.\n\nI'm currently trying to reproduce it with a minimal config.\n",
            "created_at": "2016-05-10T16:40:24Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-218216860",
            "id": 218216860,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxODIxNjg2MA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218216860/reactions"
            },
            "updated_at": "2016-05-10T16:40:24Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218216860",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/6536964?v=4",
                "events_url": "https://api.github.com/users/vccortez/events{/privacy}",
                "followers_url": "https://api.github.com/users/vccortez/followers",
                "following_url": "https://api.github.com/users/vccortez/following{/other_user}",
                "gists_url": "https://api.github.com/users/vccortez/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/vccortez",
                "id": 6536964,
                "login": "vccortez",
                "node_id": "MDQ6VXNlcjY1MzY5NjQ=",
                "organizations_url": "https://api.github.com/users/vccortez/orgs",
                "received_events_url": "https://api.github.com/users/vccortez/received_events",
                "repos_url": "https://api.github.com/users/vccortez/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/vccortez/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/vccortez/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/vccortez",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I've been adding my plugins one by one and it looks like [editorconfig/editorconfig-vim](https://github.com/editorconfig/editorconfig-vim) is the one triggering this error.\n\n**Edit**: Nevermind, editorconfig has a build step I haven't done yet. I'll try that and keep checking the rest of my plugins later.\n",
            "created_at": "2016-05-10T17:32:19Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-218231606",
            "id": 218231606,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxODIzMTYwNg==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 1,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218231606/reactions"
            },
            "updated_at": "2016-05-10T17:41:22Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218231606",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/6536964?v=4",
                "events_url": "https://api.github.com/users/vccortez/events{/privacy}",
                "followers_url": "https://api.github.com/users/vccortez/followers",
                "following_url": "https://api.github.com/users/vccortez/following{/other_user}",
                "gists_url": "https://api.github.com/users/vccortez/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/vccortez",
                "id": 6536964,
                "login": "vccortez",
                "node_id": "MDQ6VXNlcjY1MzY5NjQ=",
                "organizations_url": "https://api.github.com/users/vccortez/orgs",
                "received_events_url": "https://api.github.com/users/vccortez/received_events",
                "repos_url": "https://api.github.com/users/vccortez/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/vccortez/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/vccortez/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/vccortez",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "it is probably not related to a specific plugin, looks like the python host is not loading at all. You can try `:py3 print(\"test\")`\n",
            "created_at": "2016-05-10T18:24:22Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-218246354",
            "id": 218246354,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxODI0NjM1NA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218246354/reactions"
            },
            "updated_at": "2016-05-10T18:24:33Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/218246354",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1363104?v=4",
                "events_url": "https://api.github.com/users/bfredl/events{/privacy}",
                "followers_url": "https://api.github.com/users/bfredl/followers",
                "following_url": "https://api.github.com/users/bfredl/following{/other_user}",
                "gists_url": "https://api.github.com/users/bfredl/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/bfredl",
                "id": 1363104,
                "login": "bfredl",
                "node_id": "MDQ6VXNlcjEzNjMxMDQ=",
                "organizations_url": "https://api.github.com/users/bfredl/orgs",
                "received_events_url": "https://api.github.com/users/bfredl/received_events",
                "repos_url": "https://api.github.com/users/bfredl/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/bfredl/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/bfredl/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/bfredl",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I tracked it down to one of my plugins, that uses Xlib in order to set Caps Lock to off when I quit insert mode: https://github.com/ntrrgc/dotfiles/blob/master/.vim/plugin/capslock.vim\r\n\r\nIt seems like it's crashing in Neovim for whatever reason... but I can't find an error log anywhere.",
            "created_at": "2017-02-12T17:36:08Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-279234262",
            "id": 279234262,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI3OTIzNDI2Mg==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279234262/reactions"
            },
            "updated_at": "2017-02-12T17:36:08Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279234262",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1002436?v=4",
                "events_url": "https://api.github.com/users/ntrrgc/events{/privacy}",
                "followers_url": "https://api.github.com/users/ntrrgc/followers",
                "following_url": "https://api.github.com/users/ntrrgc/following{/other_user}",
                "gists_url": "https://api.github.com/users/ntrrgc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ntrrgc",
                "id": 1002436,
                "login": "ntrrgc",
                "node_id": "MDQ6VXNlcjEwMDI0MzY=",
                "organizations_url": "https://api.github.com/users/ntrrgc/orgs",
                "received_events_url": "https://api.github.com/users/ntrrgc/received_events",
                "repos_url": "https://api.github.com/users/ntrrgc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ntrrgc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ntrrgc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ntrrgc",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "@ntrrgc Is that plugin writing to stdout? That would break the channel. We do need to provide better feedback for that situation. On the `nvim` side, we should drop invalid messages and report them, rather than closing the channel. And we should report as much identifying information as possible that's associated with each channel.",
            "created_at": "2017-02-12T18:52:01Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-279239200",
            "id": 279239200,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI3OTIzOTIwMA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279239200/reactions"
            },
            "updated_at": "2017-02-12T18:52:01Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279239200",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "After more debugging I found in `dmesg` that python2 was actually crashing with segmentation fault. Knowing that the error makes sense. It took me a while to debug it with gdb, but finally I found it to be an error in my Python code which uses ctypes... it was receiving a pointer (8 bytes in amd64) as an int (4 bytes). \r\n\r\nFor whatever reason the memory layout of Vim was such that these pointers fit in 4 bytes, but this is not the case in the standalone Python interpreter NVim uses. After fixing my code NVim no longer crashes.\r\n\r\nI guess the error message could tell something like 'The Python server process may have crashed' so that the user knows where the issue actually is.",
            "created_at": "2017-02-12T19:09:25Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-279240445",
            "id": 279240445,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI3OTI0MDQ0NQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279240445/reactions"
            },
            "updated_at": "2017-02-12T19:14:08Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279240445",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1002436?v=4",
                "events_url": "https://api.github.com/users/ntrrgc/events{/privacy}",
                "followers_url": "https://api.github.com/users/ntrrgc/followers",
                "following_url": "https://api.github.com/users/ntrrgc/following{/other_user}",
                "gists_url": "https://api.github.com/users/ntrrgc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ntrrgc",
                "id": 1002436,
                "login": "ntrrgc",
                "node_id": "MDQ6VXNlcjEwMDI0MzY=",
                "organizations_url": "https://api.github.com/users/ntrrgc/orgs",
                "received_events_url": "https://api.github.com/users/ntrrgc/received_events",
                "repos_url": "https://api.github.com/users/ntrrgc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ntrrgc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ntrrgc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ntrrgc",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "It would also be super helpful if the traceback within NVim showed the NVim command that made it crash.",
            "created_at": "2017-02-12T19:30:47Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-279241889",
            "id": 279241889,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI3OTI0MTg4OQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279241889/reactions"
            },
            "updated_at": "2017-02-12T19:30:47Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/279241889",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1002436?v=4",
                "events_url": "https://api.github.com/users/ntrrgc/events{/privacy}",
                "followers_url": "https://api.github.com/users/ntrrgc/followers",
                "following_url": "https://api.github.com/users/ntrrgc/following{/other_user}",
                "gists_url": "https://api.github.com/users/ntrrgc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ntrrgc",
                "id": 1002436,
                "login": "ntrrgc",
                "node_id": "MDQ6VXNlcjEwMDI0MzY=",
                "organizations_url": "https://api.github.com/users/ntrrgc/orgs",
                "received_events_url": "https://api.github.com/users/ntrrgc/received_events",
                "repos_url": "https://api.github.com/users/ntrrgc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ntrrgc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ntrrgc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ntrrgc",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "@ntrrgc -- I'm seeing the same OP behavior, but I don't see any related messages in dmesg.  However, when I try @bfredl test `:py3 print(\"test\")`, I get `E117: Unknown function: provider#python3#Call`.  The neovim plugin works fine until something breaks after which it does the invalid channel thing, so it's not a total failure of missing a dependency.",
            "created_at": "2017-04-01T13:24:57Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-290919934",
            "id": 290919934,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI5MDkxOTkzNA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/290919934/reactions"
            },
            "updated_at": "2017-04-01T13:24:57Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/290919934",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10137?v=4",
                "events_url": "https://api.github.com/users/ghost/events{/privacy}",
                "followers_url": "https://api.github.com/users/ghost/followers",
                "following_url": "https://api.github.com/users/ghost/following{/other_user}",
                "gists_url": "https://api.github.com/users/ghost/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ghost",
                "id": 10137,
                "login": "ghost",
                "node_id": "MDQ6VXNlcjEwMTM3",
                "organizations_url": "https://api.github.com/users/ghost/orgs",
                "received_events_url": "https://api.github.com/users/ghost/received_events",
                "repos_url": "https://api.github.com/users/ghost/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ghost/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ghost/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ghost",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "see https://github.com/neovim/neovim/issues/6414#issuecomment-290928946 for a possible solution",
            "created_at": "2017-04-01T16:27:16Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-290930654",
            "id": 290930654,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI5MDkzMDY1NA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/290930654/reactions"
            },
            "updated_at": "2017-04-01T16:27:16Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/290930654",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "@justinmk, #6414 did not fix for me.  Re-installed Python libraries with the no-cache-dir, upgrade, and force-reinstall for both pip2 and pip3 and I still get the error.  `print(\"test\")` works, though.\r\n\r\nAlso: removed all plugins (from `plugin/` and `bundle/`) except notmuch (git@github.com:imain/notmuch-vim.git), and the issue still happens.  I've only triggered it on one particular action, sending email (which I believe starts with `rb_compose_send`); everything else seems to work fine, so it's likely something in that execution tree.\r\n\r\nI get the feeling that this is not an environmental issue, but something the plugin is doing.  Printing to STDOUT?  Any suggestions on how to debug this?  My go-to debugging tool is to stuff prints into the code, but if that's going to cause errors, what's my next option?",
            "created_at": "2017-04-08T10:55:01Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-292710396",
            "id": 292710396,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDI5MjcxMDM5Ng==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/292710396/reactions"
            },
            "updated_at": "2017-04-08T10:55:01Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/292710396",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10137?v=4",
                "events_url": "https://api.github.com/users/ghost/events{/privacy}",
                "followers_url": "https://api.github.com/users/ghost/followers",
                "following_url": "https://api.github.com/users/ghost/following{/other_user}",
                "gists_url": "https://api.github.com/users/ghost/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ghost",
                "id": 10137,
                "login": "ghost",
                "node_id": "MDQ6VXNlcjEwMTM3",
                "organizations_url": "https://api.github.com/users/ghost/orgs",
                "received_events_url": "https://api.github.com/users/ghost/received_events",
                "repos_url": "https://api.github.com/users/ghost/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ghost/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ghost/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ghost",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I still have the same problem. I usually run nvim in docker in Alpine - currently at 0.4.3r1. It works, including all the plugins, until something happens in one of the pluggins, and then I get those errors (or one of those mentioned in duplicate bugs) at every keystroke. Restarting nvim \"fixes\" it for a bit. Is there a way to restart just the python server or something from nvim maybe? To not lose the rest of the nvim state?",
            "created_at": "2020-07-17T01:15:36Z",
            "html_url": "https://github.com/neovim/neovim/issues/4468#issuecomment-659770017",
            "id": 659770017,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/4468",
            "node_id": "MDEyOklzc3VlQ29tbWVudDY1OTc3MDAxNw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/659770017/reactions"
            },
            "updated_at": "2020-07-17T01:15:36Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/659770017",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/455541?v=4",
                "events_url": "https://api.github.com/users/yohcop/events{/privacy}",
                "followers_url": "https://api.github.com/users/yohcop/followers",
                "following_url": "https://api.github.com/users/yohcop/following{/other_user}",
                "gists_url": "https://api.github.com/users/yohcop/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/yohcop",
                "id": 455541,
                "login": "yohcop",
                "node_id": "MDQ6VXNlcjQ1NTU0MQ==",
                "organizations_url": "https://api.github.com/users/yohcop/orgs",
                "received_events_url": "https://api.github.com/users/yohcop/received_events",
                "repos_url": "https://api.github.com/users/yohcop/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/yohcop/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/yohcop/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/yohcop",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 15,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/4468/comments",
    "created_at": "2016-03-18T11:14:14Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/4468/events",
    "html_url": "https://github.com/neovim/neovim/issues/4468",
    "id": 141838441,
    "labels": [
        {
            "color": "c5def5",
            "default": false,
            "description": "libnvim, Nvim RPC API",
            "id": 103819671,
            "name": "api",
            "node_id": "MDU6TGFiZWwxMDM4MTk2NzE=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/api"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 144672485,
            "name": "provider",
            "node_id": "MDU6TGFiZWwxNDQ2NzI0ODU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/provider"
        },
        {
            "color": "c7def8",
            "default": false,
            "description": "user experience",
            "id": 151718455,
            "name": "ux",
            "node_id": "MDU6TGFiZWwxNTE3MTg0NTU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/ux"
        },
        {
            "color": "c7def8",
            "default": false,
            "description": "channels, RPC, msgpack",
            "id": 242522707,
            "name": "channels-rpc",
            "node_id": "MDU6TGFiZWwyNDI1MjI3MDc=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/channels-rpc"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": "remote UI, --remote commands, p2p / peer-to-peer",
            "id": 2269137131,
            "name": "remote",
            "node_id": "MDU6TGFiZWwyMjY5MTM3MTMx",
            "url": "https://api.github.com/repos/neovim/neovim/labels/remote"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/4468/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": null,
        "closed_issues": 696,
        "created_at": "2014-05-10T20:43:04Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
            "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
            "followers_url": "https://api.github.com/users/justinmk/followers",
            "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
            "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/justinmk",
            "id": 1359421,
            "login": "justinmk",
            "node_id": "MDQ6VXNlcjEzNTk0MjE=",
            "organizations_url": "https://api.github.com/users/justinmk/orgs",
            "received_events_url": "https://api.github.com/users/justinmk/received_events",
            "repos_url": "https://api.github.com/users/justinmk/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/justinmk",
            "user_view_type": "public"
        },
        "description": "Low priority items. We plan to work on this but don't have a target date.",
        "due_on": null,
        "html_url": "https://github.com/neovim/neovim/milestone/6",
        "id": 655037,
        "labels_url": "https://api.github.com/repos/neovim/neovim/milestones/6/labels",
        "node_id": "MDk6TWlsZXN0b25lNjU1MDM3",
        "number": 6,
        "open_issues": 635,
        "state": "open",
        "title": "backlog",
        "updated_at": "2025-02-17T18:55:16Z",
        "url": "https://api.github.com/repos/neovim/neovim/milestones/6"
    },
    "node_id": "MDU6SXNzdWUxNDE4Mzg0NDE=",
    "number": 4468,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/4468/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/4468/timeline",
    "title": "Unhelpful 'Channel was closed by client'",
    "updated_at": "2025-02-16T20:50:38Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/4468",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1002436?v=4",
        "events_url": "https://api.github.com/users/ntrrgc/events{/privacy}",
        "followers_url": "https://api.github.com/users/ntrrgc/followers",
        "following_url": "https://api.github.com/users/ntrrgc/following{/other_user}",
        "gists_url": "https://api.github.com/users/ntrrgc/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/ntrrgc",
        "id": 1002436,
        "login": "ntrrgc",
        "node_id": "MDQ6VXNlcjEwMDI0MzY=",
        "organizations_url": "https://api.github.com/users/ntrrgc/orgs",
        "received_events_url": "https://api.github.com/users/ntrrgc/received_events",
        "repos_url": "https://api.github.com/users/ntrrgc/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/ntrrgc/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/ntrrgc/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/ntrrgc",
        "user_view_type": "public"
    }
}